src/dynamo/models/package.ts

```diff
@@ -0,0 +1,278 @@
+const statusAttr: keyof UnhydratedCardPackage = 'status';
+const ownerAttr: keyof UnhydratedCardPackage = 'owner';
+const keywordsAttr: keyof UnhydratedCardPackage = 'keywords';
+
+const createHydratedPackage = (
+  document: UnhydratedCardPackage,
+  owner: UserType, //TODO: User type
+  cards: CardDetails[],
+): CardPackage => {
+  return {
+    id: document.id!,
+    title: document.title,
+    date: document.date,
+    owner,
+    status: document.status,
+    cards: cards,
+    keywords: document.keywords,
+    voters: document.voters,
+    votecount: document.votecount,
+  } as CardPackage;
+};
+
+const hydrate = async (pack?: UnhydratedCardPackage): Promise<CardPackage | undefined> => {
+  if (!pack) {
+    return pack;
+  }
+
+  const owner = await User.getById(pack.owner);
+  const cards = pack.cards.map((c) => {
+    return cardFromId(c);
+  });
+
+  return createHydratedPackage(pack, owner, cards);
+};
+
+const batchHydrate = async (packs: UnhydratedCardPackage[]): Promise<CardPackage[]> => {
+  const owners: UserType[] = await User.batchGet(packs.map((pack) => pack.owner));
+
+  return packs.map((pack) => {
+    const owner = owners.find((owner) => owner.id === pack.owner);
+    const cards = pack.cards.map((c) => {
+      return cardFromId(c);
+    });
+
+    //Technically it is possible to not find an owner but let's assume our data is correct
+    return createHydratedPackage(pack, owner!, cards);
+  });
+};
+
+const applyKeywordFilter = (query: QueryInput, keywords: string): QueryInput => {
+  if (!keywords) {
+    return query;
+  }
+
+  const words = keywords?.toLowerCase()?.split(' ') || [];
+
+  // all words must exist in the keywords
+  query.FilterExpression = words.map((word) => `contains(#keywords, :${word})`).join(' and ');
+
+  query.ExpressionAttributeNames = {
+    ...query.ExpressionAttributeNames,
+    '#keywords': keywordsAttr,
+  };
+
+  query.ExpressionAttributeValues = {
+    ...query.ExpressionAttributeValues,
+    ...words.reduce((acc: Record<string, string>, word) => {
+      acc[`:${word}`] = word;
+      return acc;
+    }, {}),
+  };
+
+  return query;
+};
+
+type QueryResponse = { items?: CardPackage[]; lastKey?: DocumentClient.Key };
+
+const packages = {
+  getById: async (id: string): Promise<CardPackage | undefined> =>
+    hydrate((await client.get(id)).Item as UnhydratedCardPackage),
+  put: async (document: UnhydratedCardPackage | CardPackage): Promise<string> => {
+    const id = document.id || uuidv4();
+
+    let ownerId: string | undefined;
+    if (document.owner && typeof document.owner !== 'string' && document.owner.id) {
+      ownerId = document.owner.id;
+    } else if (document.owner && typeof document.owner === 'string') {
+      ownerId = document.owner;
+    }
+
+    let cardIds: string[] = [];
+    if (document.cards) {
+      cardIds = document.cards.map((card) => {
+        if (typeof card !== 'string' && card.scryfall_id) {
```

Comment:
In comparison when saving a package it might be a hydrated version, such as when voting/unvoting, so have to handle the cards list being list of string or list of CardDetails